### PR TITLE
Update AWS SDK v2 to 2.20.89

### PR DESCRIPTION
**Pull request by wendigo**

### CODE REVIEW

**Comment by findinpath** [comment]
@trinodb/maintainers pls run the PR with secrets

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/ByteBufferAsyncRequestBody.java [hunk 1]
```diff
@@ -1,116 +0,0 @@
...
-import software.amazon.awssdk.core.internal.async.ByteArrayAsyncRequestBody;
-import software.amazon.awssdk.core.internal.util.Mimetype;
-
-import java.nio.ByteBuffer;
-import java.util.Optional;
-
-import static java.util.Objects.requireNonNull;
-
-/**
- * This class mimics the implementation of {@link ByteArrayAsyncRequestBody} except for we use a ByteBuffer
```
Comment (losipiuk) [inline plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/ByteBufferAsyncRequestBody.java:29]: This was here for a reason

Comment (losipiuk) [reply plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/ByteBufferAsyncRequestBody.java:29]: cc: @linzebing

**Review by losipiuk — CHANGES_REQUESTED** [review]
Restore ByteBufferAsyncRequestBody.

BTW: Even if we wanted to drop it - it should be separate commit - not SDK version update

**Comment by wendigo** [comment]
@losipiuk extracted to separate commit

**Comment by wendigo** [comment]
There is also pending PR which seems relevant here: https://github.com/aws/aws-sdk-java-v2/pull/4096

File: plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/ByteBufferAsyncRequestBody.java (hunk 1 above)
Comment (losipiuk) [reply plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/s3/ByteBufferAsyncRequestBody.java:29]: Now when you changed `fromByteBufferUnsafe` it should be fine. 
@linzebing PTAL.

**Review by linzebing — APPROVED** [review]

**Comment by losipiuk** [comment]
/test-with-secrets sha=7d308deed56c6cc619d16b1b7a18a7713972e586

**Comment by electrum** [comment]
Looks like we need to squash the commits, since the `internal` class `ByteArrayAsyncRequestBody` is removed in the new version, and thus the first commit won't compile on its own.

**Pull request merged**